### DISCUSSION TITLE
piper-tts on raspberry pi 3 using Python version 3.9

### DISCUSSION THREAD

**Discussion by silvermole** [rhasspy/piper#332] [Q&A]
For a science fair project. we are installing piper-tts on a raspberry pi 3 but we are getting an error regarding ONNX Runtime. I do not have it installed. how can I resolved this error?

**Comment by silvermole** [comment]
i@raspberrypi:~ $ pip install piper-tts
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Collecting piper-tts
  Using cached piper_tts-1.2.0-py3-none-any.whl (29 kB)
  Using cached piper_tts-1.1.0-py3-none-any.whl (27 kB)
ERROR: Cannot install piper-tts==1.1.0 and piper-tts==1.2.0 because these package versions have conflicting dependencies.

The conflict is caused by:
    piper-tts 1.2.0 depends on piper-phonemize~=1.1.0
    piper-tts 1.1.0 depends on piper-phonemize~=1.0.0

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies
pi@raspberrypi:~ $

**Comment by synesthesiam** [comment]
You may need to `pip3 install -f https://synesthesiam.github.io/prebuilt-apps/ ...` to get a 32-bit wheel for the onnx runtime.

**Reply by silvermole** [reply]
Hi Michael,

I wanted to thank you for your help. I have a question regarding the prebuilt app that I will download on my Raspberry Pi 3. Do I need to download any additional Python packages? Also, could you please tell me how I can invoke this app in my Python code? I have a Python program which controls various devices like cameras, speakers, LEDs, and sensors, so I'm hoping to integrate this app with it.

If you have any Python code samples that utilize the prebuilt app you mentioned previously, please share them with me. I'm specifically looking for pre-downloaded models based TTS that work on a Raspberry Pi 3.

Thanks again for your help!

**Comment by silvermole** [comment]
Hi Michael,

I wanted to thank you for your help. I have a question regarding the prebuilt app that I will download on my Raspberry Pi 3. Do I need to download any additional Python packages? Also, could you please tell me how I can invoke this app in my Python code? I have a Python program which controls various devices like cameras, speakers, LEDs, and sensors, so I'm hoping to integrate this app with it.

If you have any Python code samples that utilize the prebuilt app you mentioned previously, please share them with me. I'm specifically looking for pre-downloaded models based TTS that work on a Raspberry Pi 3.

Thanks again for your help!

Best regards,

________________________________
